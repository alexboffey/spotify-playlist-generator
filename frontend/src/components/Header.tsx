import React from "react";
import { Divider } from "antd";

const Header: React.FunctionComponent<{ title: string }> = ({ title }) => {
  return (
    <React.Fragment>
      <header
        style={{ display: "flex", alignItems: "center", marginTop: ".5rem" }}
      >
        <h2 style={{ margin: 0 }}>{title}</h2>
      </header>
      <Divider />
    </React.Fragment>
  );
};

export default Header;
