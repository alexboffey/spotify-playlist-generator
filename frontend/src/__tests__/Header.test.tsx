import React from "react";
import { shallow } from "enzyme";
import Header from "../components/Header";

describe("<Header />", () => {
  it("Should render without the action prop", () => {
    const wrapper = shallow(<Header title="Some Title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render with the action prop", () => {
    const wrapper = shallow(
      <Header title="Some Title" action={<span>I am an action</span>} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
