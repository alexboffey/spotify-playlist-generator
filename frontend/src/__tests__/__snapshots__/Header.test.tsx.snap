// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render with the action prop 1`] = `
<Fragment>
  <header
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginTop": ".5rem",
      }
    }
  >
    <h2
      style={
        Object {
          "marginBottom": 0,
          "marginRight": "auto",
        }
      }
    >
      Some Title
    </h2>
    <span>
      I am an action
    </span>
  </header>
  <Divider />
</Fragment>
`;

exports[`<Header /> Should render without the action prop 1`] = `
<Fragment>
  <header
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginTop": ".5rem",
      }
    }
  >
    <h2
      style={
        Object {
          "marginBottom": 0,
          "marginRight": "auto",
        }
      }
    >
      Some Title
    </h2>
  </header>
  <Divider />
</Fragment>
`;
